src/simulation/LoadGenerator.cpp

```diff
@@ -700,12 +752,32 @@ LoadGenerator::TxMetrics::report()
 }
 
 TransactionFramePtr
-LoadGenerator::createTransactionFramePtr(TestAccountPtr from,
-                                         std::vector<Operation> ops,
-                                         LoadGenMode mode)
+LoadGenerator::createTransactionFramePtr(
+    TestAccountPtr from, std::vector<Operation> ops, LoadGenMode mode,
+    std::optional<uint32_t> maxGeneratedFeeRate)
 {
-    auto txf = from->tx(ops);
+    int fee = 0;
 
+    if (maxGeneratedFeeRate)
+    {
+        auto baseFee = mApp.getLedgerManager().getLastTxFee();
+        if (baseFee > *maxGeneratedFeeRate)
+        {
+            throw std::runtime_error(
+                "maxfeerate is smaller than minimum base fee");
+        }
+        auto feeRateDistr =
+            uniform_int_distribution<uint32_t>(baseFee, *maxGeneratedFeeRate);
+        // Add a bit more fee to get non-integer fee rates, such that
+        // `ceil(fee / ops.size()) == feeRate`, but
```

Comment:
should be `feeRate + 1`, right?

Reply:
Actually it should be `floor` instead (when added value is 0 `ceil` would be equal to `feeRate`, so `+ 1` won't be correct).